x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/reducer.ts

```diff
@@ -126,5 +148,26 @@ export const policyDetailsReducer: ImmutableReducer<PolicyDetailsState, AppActio
     return newState;
   }
 
+  if (action.type === 'userChangedAntivirusRegistration') {
```

Comment:
Is there a reason why you created a separate action for antivirus registration specifically? If I am reading this right, it seems like the `antivirus_registration` is just another key within the policy config. The action `userChangedPolicyConfig` should be able to do the same thing. ex) https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/management/pages/policy/view/policy_forms/protections/malware.tsx#L53

Reply:
I will bring this up during next week office hours. General thought was to avoid components being too smart and knowing how to update properties in policy config and rather keep the logic out (specifically move it to redux). What are your thoughts on this?

I noticed that due to this logic living in component we clone policy twice and have some ts-ignore directives because the code violates immutability.

Reply:
ooh okie, i can see the value in moving logic into redux, i guess im just wondering if its possible to make it more robust since i feel like policy details will only get bigger in the future?